### ISSUE TITLE
Roughness import to SWMM

### ISSUE THREAD

**Issue by TriformMoritz** [QGEP/QGEP#728]
In accordance to:
https://github.com/QGEP/QGEP/issues/727 
The roughness is imported like this in SWMM:  COALESCE(re.wall_roughness, 0.01) AS roughness,

I propose to optimize the import to SWMM:.
- Give the opportunity to use the friction (from K_COLEBROOK or coefficient de friction) and not only the roughness (K_STRICKLER)
- Adopt the default value to 0.01333, which is close to 75 m1/3/s in accordance to 4.1.3.3 of the SIA 190.
-

**Comment by sjib** [comment]
Add to user friendliness list: https://github.com/QGEP/QGEP/issues/614

**Comment by sjib** [comment]
Suggested change in 04_vw_swmm_conduits.sql: https://github.com/QGEP/datamodel/blob/95053152f858c22cd3d0d7e569c90eb45d704dfa/swmm_views/04_vw_swmm_conduits.sql#L19

@tproduit @DflGruBoe @urskaufmann - any hesitations to change this default value if no other value is given?

**Comment by urskaufmann** [comment]
I never used this attributs till now. In MOUSE - MIKE URBAN, the coefficient_of_friction is defined over the reach-material. You can define a different value for each reach - but where should I take this information from? (e.g. if there is a reliner).
I use this different coefficient of friction for each reach in MikeUrban in case of steep reaches to have a reduction because of the "Luftaufnahme". 
Do not forget: correct value for the clear_height is as (or more) important than the coefficient_of_friction. The normal clear_heigth-value for plastic is not the correct hydraulic clear_height.

**Comment by tproduit** [comment]
**Proposition from technical user group**

Adaptation of the export function
1. User interface: let the user choose between K_COLEBROOK or K_STRICKLER for the SWMM export
2. If a coefficient is provided: uses it
3. If no value is provided: use a lookup table (default coefficients for each materials)

Write a coefficient import function
-  Reimport used coefficients: Choices: _Overwrite existing coefficients_ or _Replace null coefficients_

**Comment by ponceta** [comment]
@TriformMoritz According to new 1.6.0 release, is this fixed?

**Comment by TriformMoritz** [comment]
I'll check this week and let you know.

**Comment by TriformMoritz** [comment]
Do I need to update the database version?
Because I cant. I should update my pum version, but that is not automatically possible in windows environnement (the install button in the admin tool does not work and there is no mention in the documention how to install or update pum in windows).

**Comment by sjib** [comment]
@TriformMoritz Here you can find [how to install python libraries in windows using the osgeo4w shell](https://qgep.github.io/docs/admin-guide/interlis-io/index.html#python-libraries) 

`pip install --upgrade pip` and similar for pum
`pip install --upgrade pum`

The link to the **Osgeo4W shell** is in your QGIS Folder - choose the one of the QGIS version you are using

![osgeo4wshell](https://user-images.githubusercontent.com/1265865/225897235-858def8f-7d0b-4bcc-9243-37829c4c3d8b.png)


If you are having problems please document on this issue: https://github.com/QGEP/QGEP/issues/805 or open a new one if diffrent.

Maybe you have to restart QGIS or even your computer for installation to complete.

**Comment by TriformMoritz** [comment]
Hi Stefan, would be nice if it was that easy. But I will have to do:
1. First of all I will have to update qgis 3.16. That is not easy.
- It was installed with the 64bit installer, which does not exist anymore. So the knew one I have to tell the old directory C;/Osgeo4W64 and hope it will work (lot of dll problems last time).
2. Install Pum upgrade. 
- Does not work via the osgeo4W installer (only version 0.9.12 available)
- pip3 does not work in Osgeo4W shell
- In the command line from windows pip3 gets me a ssl error: WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available. -> I will have to set some things straight in the windows registry!
3. Then I can try to upgrade my database (hope it works)
4. And the Test SWMM

**Comment by TriformMoritz** [comment]
I got it to work and tested (and lost all my comments by closing the tab so maybe this one is shorter than the one I prepared)

Default roughness set to 0.01 mm, so Strickler of 100. Seems high to me, the SIA 190 table 4 recommands values from 70 - 80)
![image](https://user-images.githubusercontent.com/33837122/226383595-4c1b02ff-93a4-444c-9ee7-72422b0dd409.png)

**Comment by TriformMoritz** [comment]
Tested Stickler  in one pipe -> conversion seems to work

**Comment by TriformMoritz** [comment]
Tested Colebrook, conversion is not working.  use this formula from ATV_DWK-A110E 4*RACINE(g)*(32/D)^(1/6)*LOG(3.7*D/(K/1000);10) 
![image](https://user-images.githubusercontent.com/33837122/226385644-b0932224-f838-4567-865f-edc66a65ab07.png)

**Comment by TriformMoritz** [comment]
Tested Strickler and Colebrook -> Strickler is choosen. I do not have (or find) an option to choose Colebrook over Strickler.

**Comment by sjib** [comment]
> The link in the description does not work: https://qgep.github.io/docs/qgep_swmm/coefficient_of_friction.html

@TriformMoritz Please check this:
Final documentation is not yet merged: https://github.com/QGEP/docs/pull/198

**Comment by tproduit** [comment]
> Tested Colebrook, conversion is not working. use this formula from ATV_DWK-A110E 4_RACINE(g)_(32/D)^(1/6)_LOG(3.7_D/(K/1000);10) ![image](https://user-images.githubusercontent.com/33837122/226385644-b0932224-f838-4567-865f-edc66a65ab07.png)

I will double check the formula and let you know.

**Comment by tproduit** [comment]
@TriformMoritz we probably interpreted wrongly the formula. To convert from K_strickler (wall_roughness) > K_colebrook (coefficient_of_friction) we have to invert your formula.

For a diameter of  1250mm and k_st 91, we would have:

`SELECT (1000. * 3.71 * 1250. / 1000.)/(POWER(10., 91./( 4. * SQRT(9.81) * POWER(32. / (1250. /1000.), 1./6.)))) as k_colebrook`

Then we will have to take `1/k_colebrook` for swmm.

Can you confirm?

**Comment by tproduit** [comment]
> Tested Strickler and Colebrook -> Strickler is choosen. I do not have (or find) an option to choose Colebrook over Strickler.

Indeed, we finally decided (with @sjib  to not have an option to choose which coefficient is exported but rather have a priority:
	1. Coeffficient_of_friction
	2. Wall_roughness
	3. Default_coefficient_of_friction filled with lookup table
	4. General default value
